utils/time_slots_utils.js

```diff
@@ -0,0 +1,84 @@
+const {models} = require('../db/relations')
+const TimeSlot = models.TimeSlot
+
+const TimeSlotsUtils = {
+
+    getIntersection: function (slots) {
+        let res = []
+        let pointers = new Array(slots.length).fill(0)
+        let l = -1
+        let r = -1
+        while (!isNaN(l) && !isNaN(r) ) {
+            let it_slots = new Array(slots.length).fill(0)
+            for (let i = 0; i < slots.length; i++) {
+                it_slots[i] = slots[i][pointers[i]]
+            }
+            let vals = it_slots.map(slot => slot !== undefined ? slot.startDatetime.getTime() : undefined)
+            l = Math.max(...vals)
+
+            vals = it_slots.map(slot => slot !== undefined ? slot.startDatetime.getTime() + slot.duration * 60000 : undefined)
+            r = Math.min(...vals)
+
+            if (l + 15 <= r) {
```

Comment:
Jeżeli 15 to minimalne duration, to nazwałbym to jakoś, żeby było to choć trochę czytelniejsze XD